boot/bootsystemstate20.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package boot
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/bootloader"
+)
+
+// bootSystemState20 implements the bootSystemState and bootStateUpdate
+// interfaces. It is used for selecting the recovery system and mode for the
+// subsequent boot.
+type bootSystemState20 struct {
+	bl           bootloader.Bootloader
+	system, mode string
+}
+
+func newBootSystemState20() bootSystemState {
+	return &bootSystemState20{}
+}
+
+func (b *bootSystemState20) setSystemMode(system, mode string) (bootStateUpdate, error) {
+	if b.bl == nil {
+		opts := &bootloader.Options{
+			// setup the recovery bootloader
+			Recovery: true,
+		}
+		bl, err := bootloader.Find(InitramfsUbuntuSeedDir, opts)
```

Comment:
@mvo5 @bboozzoo   should we actually keep that directory around in run mode though? or should we unmount it and need to remount only when needed?   is /var/lib/snapd/seeded at least a RO mount ?

Reply:
Just checked, ubuntu-core-initramfs mounts /run/mnt/ubuntu-seed in RW mode. The core core20 handle-writable-paths sets up a fstab entry that binds mount it to /var/lib/snapd/seed, but still RW mode.

Reply:
@bboozzoo can you create the bug xnox asked about at the writable-path bits?

Reply:
Added https://bugs.launchpad.net/snapd/+bug/1869743

Reply:
I don't think we should carry around that directory in run-mode IMHO. OTOH, I don't know where the code should look for to find the ubuntu-seed bootloader, so perhaps it does need to be in this location. However, if we want ubuntu-seed to be in this location, then I think we should rename the InitramfsUbuntuSeedDir to something that isn't specific to the initramfs because this function will be used from places not during the initramfs.